dao-contracts/tests/test_admin.rs

```diff
@@ -0,0 +1,125 @@
+mod governance_voting_common;
+extern crate speculate;
+use speculate::speculate;
+use std::time::Duration;
+
+use casper_dao_contracts::{
+    action::Action,
+    voting::{voting::Voting, VotingContractCreated, VotingId},
+    AdminContractTest,
+};
+
+use casper_dao_utils::{Address, TestEnv};
+use casper_types::U256;
+
+speculate! {
+    context "admin" {
+        before {
+            let informal_quorum = 500.into();
+            let formal_quorum = 750.into();
+            let minimum_reputation = 500.into();
+            let reputation_to_mint = 10000;
+            let informal_voting_time: u64 = 3600;
+            let formal_voting_time: u64 = 2*3600;
+            let env = TestEnv::new();
+            let mut variable_repo_contract = governance_voting_common::get_variable_repo_contract(&env, informal_quorum, formal_quorum, informal_voting_time, formal_voting_time, minimum_reputation);
+            let mut reputation_token_contract = governance_voting_common::get_reputation_token_contract(&env, reputation_to_mint);
+
+            #[allow(unused_mut)]
+            let mut admin_contract = AdminContractTest::new(
+                &env,
+                Address::from(variable_repo_contract.get_package_hash()),
+                Address::from(reputation_token_contract.get_package_hash()),
+            );
+
+            variable_repo_contract
+                .add_to_whitelist(admin_contract.address())
+                .unwrap();
+
+            reputation_token_contract
+                .add_to_whitelist(admin_contract.address())
+                .unwrap();
+
+            reputation_token_contract.change_ownership(admin_contract.address()).unwrap();
+        }
+
+        test "that repo voter has been set up correctly" {
+            assert_eq!(admin_contract.get_reputation_token_address(), reputation_token_contract.address());
+            assert_eq!(admin_contract.get_variable_repo_address(), variable_repo_contract.address());
+
+            admin_contract.assert_last_event(
+                VotingContractCreated {
+                    variable_repo: Address::from(variable_repo_contract.get_package_hash()),
+                    reputation_token: Address::from(reputation_token_contract.get_package_hash()),
+                    voter_contract: Address::from(admin_contract.get_package_hash()),
+                },
+            );
+        }
+
+        context "voting" {
+            before {
+                admin_contract
+                .create_voting(
+                    reputation_token_contract.address(),
+                    Action::AddToWhitelist,
+                    env.get_account(1),
```

Comment:
I would name `env.get_account(1)` in the `before` section. Generally, I would set names for all the actors and avoid direct usage of `env.get_account(n)`